wordpress-mobile/gutenberg-mobile
# Bump Jetpack submodule and fix break changes

## Description

This PR addresses two items:
1. Bump Jetpack submodule
2. Fix break changes caused by recent updates in Jetpack's repository

### Jetpack break changes
Jetpack's repository has recently been [reorganised as a Monorepo](https://github.com/Automattic/jetpack/pull/18257), unfortunately this change has lead to a couple of break changes in the integration with the project.

### 1 - Jetpack project located in a new path
The project is now located in a different folder:
- Previously: `jetpack`
- Now: `jetpack/projects/plugins/jetpack`

This change has implied to update all the references to Jetpack that we have in the project.


### ~~2 - Haste module naming collision~~
~~The Jetpack's repository now has two `package.json` files with the same module name `Jetpack`. This is causing `Metro` bundler to fail when the bundle is generated:~~
```
jest-haste-map: Haste module naming collision: Jetpack
  The following files share their name; please adjust your hasteImpl:
    * <rootDir>/jetpack/projects/plugins/jetpack/package.json
    * <rootDir>/jetpack/package.json
Failed to construct transformer:  DuplicateError: Duplicated files or mocks. Please check the console for more info
```

~~For now as a workaround, the file located at root level (`jetpack/package.json`) has been ignored in the Metro configuration.~~

~~**However this change won't be necessary if the name is changed on one of the `package.json` files. Jetpack team is taking a look into it so maybe this change is removed in the future.**~~

This issue has been fixed in a [PR](https://github.com/Automattic/jetpack/pull/18367) from Jetpack's repository.

## Test
1. Run `npm install`.
2. Run `npm run bundle`.
3. Check that the bundles are generated properly and no error has been raised.
4. Run `npm run test`.
5. Verify that unit tests succeed.

PR submission checklist:
